src/main/java/com/kaytat/simpleprotocolplayer/MusicService.java

```diff
@@ -222,253 +197,33 @@ void configVolume() {
-                                throw new Exception("Not running");
-                            }
-                        }
-                    }
-
-                    // Get a packet
-                    is.readFully(byteArray[idx]);
-                    idx++;
-                    if (idx == NUM_PKTS) {
-                        idx = 0;
-                    }
-                    synchronized (filledLock) {
-                        filled++;
-                        if (filled == NUM_PKTS) {
-                            Log.i(TAG, "flushing");
-                            // Filled up.  Throw away everything that's in the network queue.
-                            int rd;
-                            do {
-                                rd = is.read(tmpBuf);
-                            } while (rd == packet_size && running);
-                        }
-                        filledLock.notifyAll();
-                    }
-                }
-            }
-            catch (Exception e) {
-                Log.i(TAG, "exception:" + e);
-            }
-
-            try {
-                if (socket != null) {
-                    socket.close();
-                }
-                if (running) {
-                    // Broke out of loop unexpectedly.  Shutdown.
-                    Handler h = new Handler(context.getMainLooper());
-                    Runnable r = new Runnable() {
-                        @Override
-                        public void run() {
-                            Toast.makeText(getApplicationContext(), "Unable to stream", Toast.LENGTH_SHORT).show();
-                            processStopRequest();
-                        }
-                    };
-                    h.post(r);
-                }
-            }
-            catch (Exception e) {
-                Log.i(TAG, "exception while closing:" + e);
-            }
-
-            Log.i(TAG, "done");
         }
     }
 
     /**
      * Play the stream using the given IP address and port
      */
-    void playStream(String serverAddr,
-                    int serverPort,
-                    int sample_rate,
-                    boolean stereo,
-                    int buffer_ms) {
-        int format = stereo ?
-                AudioFormat.CHANNEL_OUT_STEREO :
-                AudioFormat.CHANNEL_OUT_MONO;
-
-        // Sanitize input, just in case
-        if (sample_rate <= 0) {
-            sample_rate = DEFAULT_SAMPLE_RATE;
-        }
+    void playStream(String serverAddr, int serverPort, int sample_rate,
+            boolean stereo, int buffer_ms) {
 
-        if (buffer_ms <= 5) {
-            buffer_ms = DEFAULT_BUFFER_MS;
-        }
-
-        int minBuf = AudioTrack.getMinBufferSize(
-                sample_rate,
-                format,
-                AudioFormat.ENCODING_PCM_16BIT);
 
-        initNetworkData(sample_rate, stereo, minBuf, buffer_ms);
-
-        mState = State.Stopped;
         relaxResources();
```

Comment:
relaxResources stop all existing streams.  So when you add the new one, the old ones remain stopped?